graphql_compiler/compiler/expressions.py

```diff
@@ -856,6 +858,7 @@ def to_gremlin(self):
         immediate_operator_format = u'({left} {operator} {right})'
         dotted_operator_format = u'{left}.{operator}({right})'
         intersects_operator_format = u'(!{left}.{operator}({right}).empty)'
+        negated_dotted_operator_format = u'!{left}.{operator}({right})'
```

Comment:
The `intersects_operator_format` and the `negated_dotted_operator_format` seem fairly similar. Do you think it makes sense to deduplicate their content and express one in terms of the other?

Reply:
I'm not exactly sure intersects is like negation-- we might've also implemented this with no bang by checking if the size is nonzero (assuming there's a way to get size of the results), so I suspect it might just be a coincidence.

I think if we run into more cases of where we have the bang in front (e.g. it turns out to be useful to have a filter for non-empty lists or something, in which case you check if it's not empty), then we could look into deduplicating it.